ddtrace/internal/utils/http.py

```diff
@@ -38,6 +40,60 @@ def strip_query_string(url):
     return h + fs + f
 
 
+def redact_query_string(query_string, query_string_obfuscation_pattern):
+    # type: (str, Optional[re.Pattern]) -> Union[bytes, str]
+    if query_string_obfuscation_pattern is None:
+        return query_string
+
+    bytes_query = query_string if isinstance(query_string, bytes) else query_string.encode("utf-8")
+    return query_string_obfuscation_pattern.sub(b"<redacted>", bytes_query)
+
+
+def redact_url(url, query_string_obfuscation_pattern, query_string=None):
+    # type: (str, re.Pattern, Optional[str]) -> Union[str,bytes]
+
+    # Avoid further processing if obfuscation is disabled
+    if query_string_obfuscation_pattern is None:
+        return url
+
+    parts = compat.parse.urlparse(url)
+    redacted_query = None
+
+    if query_string:
+        redacted_query = redact_query_string(query_string, query_string_obfuscation_pattern)
+    elif parts.query:
+        redacted_query = redact_query_string(parts.query, query_string_obfuscation_pattern)
+
+    if redacted_query is not None:
+        redacted_parts = parts[:4] + (redacted_query,) + parts[5:]  # type: Tuple[Union[str, bytes], ...]
```

Comment:
Maybe check the length of parts to ensure is > 5

Reply:
It won't break (this is Python2):
```
>>> tuple()[:5]
()
>>> tuple()[4:]
()
```

But to make sure that `redacted_parts` makes sense, let's ensure that `len(parts) >= 5`